From 1f96d1a31be0ef58e6f9380ca7fb743561e76696 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@windriver.com>
Date: Fri, 22 Mar 2013 20:46:36 -0700
Subject: [PATCH 04/10] ck: set idleprio tasks to nice 19 for vm reclaim

Imported from: http://ck.kolivas.org/patches/3.0/3.8/3.8-ck1/patches/mm-idleprio_prio-1.patch

Set the effective priority of idleprio tasks to that of nice 19 tasks when
modifying vm reclaim behaviour.

-ck

Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 include/uapi/linux/sched.h |    2 +-
 mm/vmscan.c                |    2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/uapi/linux/sched.h b/include/uapi/linux/sched.h
index 00a524e..c2ebb11 100644
--- a/include/uapi/linux/sched.h
+++ b/include/uapi/linux/sched.h
@@ -39,9 +39,9 @@
 #define SCHED_BATCH		3
 /* SCHED_ISO: Implemented on BFS only */
 #define SCHED_IDLE		5
+#define SCHED_IDLEPRIO		SCHED_IDLE
 #ifdef CONFIG_SCHED_BFS
 #define SCHED_ISO		4
-#define SCHED_IDLEPRIO		SCHED_IDLE
 #define SCHED_MAX		(SCHED_IDLEPRIO)
 #define SCHED_RANGE(policy)	((policy) <= SCHED_MAX)
 #endif
diff --git a/mm/vmscan.c b/mm/vmscan.c
index a224904..98f569e 100644
--- a/mm/vmscan.c
+++ b/mm/vmscan.c
@@ -1991,6 +1991,8 @@ static inline int effective_sc_prio(struct task_struct *p)
 	if (likely(p->mm)) {
 		if (rt_task(p))
 			return -20;
+		if (p->policy == SCHED_IDLEPRIO)
+			return 19;
 		return task_nice(p);
 	}
 	return 0;
-- 
1.7.10.4

